File: CF-contest/875/B_Array_merging.py
t = int(input())
for _ in range(t):
    n = int(input())
    ar1 = list(map(int,input().split()))
    ar2 = list(map(int,input().split()))
    mp1 = {}
    for i in range(n):
        tmp = ar1[i]
        j=i
        cnt=0
        while(j<n and ar1[j]==tmp):
            cnt+=1
            j+=1
        if tmp not in mp1: mp1[tmp]=cnt
        else: mp1[tmp] = max(mp1[tmp],cnt)
        i=j
    
    mp2 = {}
    for i in range(n):
        tmp = ar2[i]
        j=i
        cnt=0
        while(j<n and ar2[j]==tmp):
            cnt+=1
            j+=1
        if tmp not in mp2: mp2[tmp]=cnt
        else: mp2[tmp] = max(mp2[tmp],cnt)
        i=j

    for key,val in mp2.items():
        if key in mp1: mp1[key]+=val
        else: mp1[key]=val
    
    ans = 0
    for key,val in mp1.items():
        ans = max(ans,val)
    print(ans)

File: 875/B_Array_merging.cpp
/*----------------------------------------
    author:  Imam
    created: 25.03.2025 16:18:45
----------------------------------------*/
#include<bits/stdc++.h>
using namespace std;
inline void solve()
{
    int n; cin>>n;
    vector<int> ar1(n),ar2(n);
    for(int &i:ar1) cin>>i;
    for(int &i:ar2) cin>>i;

    unordered_map<int,int>ump1,ump2;
    for(int i=0;i<n;i++){
        int x = ar1[i], cnt=0;
        while(i<n && ar1[i]==x){cnt++;i++;}
        ump1[x] = max(ump1[x],cnt);
        --i;
    }
    for(int i=0;i<n;i++){
        int x = ar2[i], cnt=0;
        while(i<n && ar2[i]==x){cnt++;i++;}
        ump2[x] = max(ump2[x],cnt);
        --i;
    }

    for(auto [x,y]:ump2){
        if(ump1[x]){ump1[x]+=y;}
        else {ump1[x]=y;}
    }

    int ans=INT_MIN;
    for(auto [x,y]:ump1){ans=max(ans,y);}
    cout<<ans<<'\n';
}
int main()
{
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(nullptr); cout.tie(NULL);

    int t; cin>>t;
    while(t--) solve();
    return 0;
}